### PR TITLE
feat(curriculum): add stack class lab

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by Sembauke — APPROVED** [review]

**Comment by moT01** [comment]
Blocking this while we figure out if we want to use the existing ID's or new ones.

**Comment by Dario-DC** [comment]
> Blocking this while we figure out if we want to use the existing ID's or new ones.

Just to clarify, this is very similar to the old challenge with a couple of minor changes. But I guess we want to deal with all the converted challenges in the same way.

**Review by moT01 — CHANGES_REQUESTED** [review]
Yea, I think we want to use the existing ID here - `587d8250367417b2b2512c5f`

Just curious, what were the minor changes?

File: curriculum/challenges/english/25-front-end-development/lab-stack-class/66d9ac40e5c98f4dc9e979f4.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+Your `Stack` class should have a `peek` method.
+
+```js
+const test = new Stack();
+assert.isFunction(test.peek);
+```
+
+Your `Stack` class should have a `isEmpty` method.
```
Comment (zairahira) [inline curriculum/challenges/english/25-front-end-development/lab-stack-class/66d9ac40e5c98f4dc9e979f4.md:67]:
```suggestion
Your `Stack` class should have an `isEmpty` method.
```

**Comment by zairahira** [comment]
@moT01, I went through the lab and there are only some monir changes from the original:

- Instructions are added in the user stories format.
- There are a few more hints.
- The following `print()` method isn't there in the new seed code.

```js
print() {
    console.log(this.collection);
  }
```
These aren't major changes, so the challengeID should be the same as existing.

We can change `66d9ac40e5c98f4dc9e979f4` --> `587d8250367417b2b2512c5f`

**Comment by Dario-DC** [comment]
I made confusion with another lab. Here the changes are absolutely negligible.

**Review by jdwilkin4 — APPROVED** [review]

**Review by zairahira — APPROVED** [review]
LGTM 👍️

**Review by moT01 — CHANGES_REQUESTED** [review]
The filename should change to match the challenge id.

**Comment by ilenia-magoni** [comment]
you have merge conflicts

File: curriculum/challenges/_meta/lab-stack-class/meta.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "Build a Stack Class",
+  "isUpcomingChange": true,
+  "usesMultifileEditor": true,
+  "blockType": "lab",
+  "dashedName": "lab-stack-class",
+  "order": 264,
```
Comment (moT01) [inline curriculum/challenges/_meta/lab-stack-class/meta.json:7]:
```suggestion
  "order": 263,
```

Looks like the order changed for this one.

Comment (Dario-DC) [reply curriculum/challenges/_meta/lab-stack-class/meta.json]: Updated :+1:

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**